### DISCUSSION TITLE
ESP32 with 32x64 LED Matrix Issues

### DISCUSSION THREAD

**Discussion by acktarus** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#603] [Q&A]
I have a weird issue with a LED Matrix project I am working on. I was able to build a "test" project and the LED Matrix is driven perfectly. Then I added Blynk.Edgent in the mix (Wifi connection) and now there is interference on the screen. See Video. I really don't know how to get rid of this interference.

The ESP is a ESP-WROOM-32 and the LED Display is P2.5-16S-V1.0

Pinout is as following:

#define R1_PIN_DEFAULT 25
#define G1_PIN_DEFAULT 26
#define B1_PIN_DEFAULT 27
#define R2_PIN_DEFAULT 14
#define G2_PIN_DEFAULT 12
#define B2_PIN_DEFAULT 13
#define A_PIN_DEFAULT 23
#define B_PIN_DEFAULT 19
#define C_PIN_DEFAULT 5
#define D_PIN_DEFAULT 17
#define E_PIN_DEFAULT   -1 // IMPORTANT: Change to a valid pin if using a 64x64px panel.        
#define LAT_PIN_DEFAULT 4
#define OE_PIN_DEFAULT 15
#define CLK_PIN_DEFAULT 16

Here is a Video of the issue
[Video](https://photos.onedrive.com/share/AE7D877370800A57!219877?cid=AE7D877370800A57&resId=AE7D877370800A57!219877&authkey=!ACbKk460qxq3MX4&ithint=video&e=xyRvd0)

**Comment by mrcodetastic** [comment]
Are you printing stuff to serial output? This can cause issues. 

Also, don't have cables any longer than 10cm.

**Reply by acktarus** [reply]
I am indeed printing stuff to serial output (I will comment them out and see if this fixes my problem).   I am using the cable that came with the screen (I will also shorten it... but I know it didn't cause an issue when the blynk.edgent part of my code was not used)

**Reply by mrcodetastic** [reply]
Comment out the serial stuff first. That might fix it. 

Use a gpio that's further away from tx,rx pins

**Reply by acktarus** [reply]
I traced my problem to PIN 25  of my ESP32.   When I disconnect the R1 from PIN 25, the interference goes away.  

But I can't seem to be able to move the GPIO to a different pin.   Maybe I am using the wrong #Define command in my program.  I am using "#define R1_PIN_DEFAULT  25"    

I have change that number to 34 or 33 and moved my wire with no luck (the red on the top quadrant is not displaying properly

Am I missing something?
![image](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/11763253/8eb715da-d4eb-4b1e-ac22-f471828c85bb)

**Reply by mrcodetastic** [reply]
Look at the examples on how to change mapping with mxconfig variable
